store/rootmulti/store_test.go

```diff
@@ -116,7 +116,7 @@ func TestCacheMultiStoreWithVersion(t *testing.T) {
 	// require we cannot commit (write) to a cache-versioned multi-store
 	require.Panics(t, func() {
 		kvStore.Set(k, []byte("newValue"))
-		cms.Write()
+		kvStore.(types.CacheWrap).Write()
```

Comment:
I assume, you want to test for `cms.Write()` to panic. Your workaround with `kvStore.(types.CacheWrap).Write()` is pragmatic but with a `cms.Write()` you give away control from the main thread. Therefore the caller can not handle panics which is the bigger problem.
Another issue would be deterministic behaviour when more than one store panics. 
You could recover in the go routines, collect and sort the errors. Maybe it is worth to refactor to error return values instead of panics.